### DISCUSSION TITLE
How to iterate and explode all polylines in modelspace?

### DISCUSSION THREAD

**Discussion by umli187** [mozman/ezdxf#309] [Q&A] [answered]
Hello,

Is it possible to explode all polylines in .dxf document to lines and arcs? 
I would like to do something like iteration over all polyline entities use `explode(target_layout: BaseLayout = None) → EntityQuery` and save result, Could you give me short example of code? 

Thank you

**Comment by mozman** [comment, marked as answer]
Iterate POLYLINE entities in modelspace and explode them:

```Python
msp = doc.modelspace()
for entity in msp.query("POLYLINE"):
     entity.explode()

```

To do the same for LWPOLYLINE entities, replace `"POLYLINE"` by `"LWPOLYLINE"`, to iterate both entity types use the query string: 

`"POLYLINE LWPOLYLINE"`

Related documentation:
- [Usage for Beginners](https://ezdxf.mozman.at/docs/usage_for_beginners.html)
- [Tutorial for getting data from DXF files](https://ezdxf.mozman.at/docs/tutorials/getting_data.html)
- [Entity Query Reference](https://ezdxf.mozman.at/docs/query.html)
